lib/config/path-util.js

```diff
@@ -0,0 +1,110 @@
+     */
+    getContentBase(webpackConfig) {
+        // strip trailing slash (for Unix or Windows)
+        const outputPath = webpackConfig.outputPath.replace(/\/$/,'').replace(/\\$/, '');
+        // use the manifestKeyPrefix if available
+        const publicPath = webpackConfig.manifestKeyPrefix ? webpackConfig.manifestKeyPrefix.replace(/\/$/,'') : webpackConfig.publicPath.replace(/\/$/,'');
+
+        /*
+         * We use the intersection of the publicPath (or manifestKeyPrefix) and outputPath
+         * to determine the "document root" of the web server. For example:
+         *   * outputPath = /var/www/public/build
+         *   * publicPath = /build/
+         *      => contentBase should be /var/www/public
+         */
+
+        // start with outputPath, then join publicPath with it, see if it equals outputPath
+        // in loop, do dirname on outputPath and repeat
+        // eventually, you (may) get to the right path
+        let contentBase = outputPath;
+        while (path.dirname(contentBase) !== contentBase) {
+            contentBase = path.dirname(contentBase);
+
+            if (path.join(contentBase, publicPath) === outputPath) {
+                return contentBase;
+            }
+        }
+
+        throw new Error(`Unable to determine contentBase option for webpack's devServer configuration. The publicPath (${webpackConfig.publicPath}) string does not exist in the outputPath (${webpackConfig.outputPath}), and so the "document root" cannot be determined.`);
+    },
+
+    /**
+     * Returns the output path, but as a relative string (e.g. web/build)
+     *
+     * @param {WebpackConfig} webpackConfig
+     * @return {String}
+     */
+    getRelativeOutputPath(webpackConfig) {
+        return webpackConfig.outputPath.replace(webpackConfig.getContext() + path.sep, '');
+    },
+
+    /**
+     * If the manifestKeyPrefix is not set, this uses the publicPath to generate it.
+     *
+     * Most importantly, this runs some sanity checks to make sure that it's
+     * ok to use the publicPath as the manifestKeyPrefix.
+     *
+     * @param {WebpackConfig} webpackConfig
+     * @return {String}
+     */
+    generateManifestKeyPrefix(webpackConfig) {
+        if (webpackConfig.publicPath.includes('://')) {
+            /*
+             * If publicPath is absolute, you probably don't want your manifests.json
+             * keys to be prefixed with the CDN URL. Instead, we force you to
+             * choose your manifestKeyPrefix.
+             */
+
+            throw new Error('Cannot determine how to prefix the keys in manifest.json. Call Encore.setManifestKeyPrefix() to choose what path (e.g. build/) to use when building your manifest keys. This is happening because you passed an absolute URL to setPublicPath().');
+        }
+
+        let outputPath = webpackConfig.outputPath;
+        // for comparison purposes, change \ to / on Windows
+        outputPath = outputPath.split('\\').join('/');
```

Comment:
no, when using a regex, it depends whether you pass a regex matching the first place or all of them (think `preg_match` vs `preg_match_all` in PHP). I added the `g` modifier on my regex.